scripts/dts/extract/flash.py

```diff
@@ -91,6 +91,36 @@ def _extract_partition(self, node_path):
 
         insert_defs("DT_FLASH_AREA", prop_def, prop_alias)
 
+    def _extract_partition_dev(self, node_address):
+        prop_def = {}
+        prop_alias = {}
+        node = reduced[node_address]
+
+        partition_name = node['props']['label']
+        partition_sectors = node['props']['reg']
+
+        # Extract a per partition dev name, something like:
+        # #define DT_FLASH_AREA_1_DEV             "FLASH_CTRL"
+
+        # For now assume node_address is something like:
+        # /flash-controller@4001E000/flash@0/partitions/partition@fc000
+        # first we go up 2 levels to get the flash, check its compat
+        #
+        # The flash controller might be the flash itself (for cases like NOR
+        # flashes), for the case of 'soc-nv-flash' we assume its the parent
+        # of the flash node.
+        ctrl_addr = '/' + '/'.join(node_address.split('/')[1:-2])
+        if get_compat(ctrl_addr) == "soc-nv-flash":
+            ctrl_addr = '/' + '/'.join(node_address.split('/')[1:-3])
+
+        node = reduced[ctrl_addr]
+        name = "\"{}\"".format(node['props']['label'])
+
+        label = self.get_label_string(["DT_FLASH_AREA"] + [partition_name] + ["DEV",])
```

Comment:
Can be simplified to

```python
label = self.get_label_string(["DT_FLASH_AREA", partition_name, "DEV"])
```

No need to do `[foo,]` for a single-element list, though you see it in some other places. Just `[foo]` works fine.